Vibe 3

YossiSaadi opened the pull request.

talkor commented on packages/core/src/components/Tipseen/Tipseen.tsx:
```diff
@@ -219,10 +199,8 @@ const Tipseen: VibeComponent<TipseenProps> & {
 
 export default withStaticProps(Tipseen, {
-  closeButtonThemes: TipseenCloseButtonTheme,
-  positions: DialogPosition,
-  animationTypes: AnimationType,
-  justifyTypes: JustifyType,
-  hideShowTriggers: HideShowEvent,
```
@rivka-ungar  we need to bring these back, right?

talkor commented on packages/core/src/components/Tipseen/Tipseen.tsx:
```diff
@@ -58,69 +55,56 @@ export interface TipseenProps extends VibeComponentProps {
 
 const Tipseen: VibeComponent<TipseenProps> & {
-  closeButtonThemes?: typeof TipseenCloseButtonTheme;
-  positions?: typeof DialogPosition;
```
@rivka-ungar we need to bring it back, no?

talkor commented on packages/core/src/components/Menu/MenuItem/MenuItem.tsx:
```diff
@@ -72,15 +66,12 @@ export interface MenuItemTitleComponentProps extends Omit<MenuItemProps, "title"
 
 const MenuItem: VibeComponent<MenuItemProps | MenuItemTitleComponentProps> & {
   iconType?: typeof Icon.type;
-  tooltipPositions?: typeof DialogPosition;
```
@rivka-ungar we need to bring it back, no?

talkor commented on packages/core/src/components/Menu/MenuItem/MenuItem.tsx:
```diff
@@ -175,6 +159,5 @@ Object.assign(MenuItem, {
 
 export default withStaticProps(MenuItem, {
-  iconType: Icon.type,
-  tooltipPositions: DialogPosition
```
Same

talkor commented on packages/core/src/components/Menu/MenuItemButton/MenuItemButton.tsx:
```diff
@@ -142,6 +133,5 @@ Object.assign(MenuItemButton, {
 
 export default withStaticProps(MenuItemButton, {
-  kinds: Button.kinds,
-  tooltipPositions: DialogPosition
```
Same

talkor commented on packages/core/src/components/SplitButton/SplitButton.tsx:
```diff
@@ -59,8 +57,6 @@ export interface SplitButtonProps extends ButtonProps {
 
 const SplitButton: FC<SplitButtonProps> & {
-  secondaryPositions?: typeof SplitButtonSecondaryContentPosition;
-  secondaryDialogPositions?: typeof SplitButtonSecondaryContentPosition;
```
Same

talkor commented on packages/core/src/components/SplitButton/SplitButton.tsx:
```diff
@@ -290,14 +283,11 @@ SplitButton.defaultProps = {
 export default withStaticProps(SplitButton, {
-  // Backward compatibility for enum naming
-  secondaryPositions: SplitButtonSecondaryContentPosition,
-  secondaryDialogPositions: SplitButtonSecondaryContentPosition,
```
Same

YossiSaadi commented:
Writing here so we won't forget
We need to merge this not as a squash and merge (!) otherwise we will lose tons of context

YossiSaadi commented on packages/core/src/hooks/useActiveDescendantListFocus/__stories__/useActiveDescendantListFocus.stories.tsx:
```diff
@@ -14,6 +14,7 @@ type Story = StoryObj<typeof useActiveDescendantListFocus>;
 
 export default {
   title: "Hooks/useActiveDescendantListFocus",
+  // @ts-ignore
```
how come those kind of stuff that you've added ignore in many places didn't bug us until today?

talkor replied:
Because the stories file migrated to ts
This is a bug we need to resolve but not for this scope IMO

YossiSaadi commented on packages/core/src/components/DatePicker/DatePicker.tsx:
```diff
@@ -1,7 +1,7 @@
 import cx from "classnames";
 import moment from "moment";
-import "react-dates/initialize";
+import "react-dates/initialize.js";
```
why?

talkor replied:
type: module

YossiSaadi replied:
really? :|

YossiSaadi reviewed (commented):
I cannot believe I survived this CR
Amazing work !!!
Cannot approve since I've created this PR

talkor commented on packages/core/types/files.d.ts:
```diff
@@ -12,3 +12,4 @@ declare module "*.svg" {
 
 declare module "*.png";
 declare module "*.mp4";
+declare module "*.pdf";
```
Because of the accessibility pdfs

talkor commented on packages/core/src/hooks/useSwitch/__stories__/useSwitch.mdx:
````diff
@@ -18,6 +18,12 @@ Hook for controlling boolean state on components, e.g. Toggle, by exposing state
+### Import
+
+```js
+import { # useSwitch } from "@vibe/core/hooks";
````
Lol good catch

talkor commented on .github/workflows/chromatic.yml:
```diff
@@ -35,3 +35,4 @@ jobs:
           projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
           workingDir: packages/core
           exitOnceUploaded: true
+          onlyChanged: false
```
Done https://monday.monday.com/boards/3532714909/pulses/7773587603

talkor commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+
+We use [Prettier](https://prettier.io/) for code formatting. Please, make sure to use it in your editor to keep the codebase consistent.
+
+To run linting and formatting checks, use:
```
See line 24

talkor commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,63 @@
+- Link to any related issues or discussions
+- Make sure that all checks are passed
+
+After submitting your PR, the maintainers will review your changes and provide feedback. If everything is good, your PR will be merged.
```
You sure? I don't think it's really informative

YossiSaadi replied:
Whatever you prefer
I think if we're auto-assigned, they don't need to tell us "I've created a PR" in the issue

talkor commented on packages/codemod/README.md:
````diff
@@ -0,0 +1,70 @@
+  npx @vibe/codemod --target /path/to/your/project
+  ```
+
+### `--extensions` (alias: `-x`)
+
+- **Description**: Specifies which file extensions to include for the migration.
+- **Choices**: `jsx`, `tsx`, `js`, `ts`
+- **Required**: Yes
+- **Example**:
+
+  ```bash
+  npx @vibe/codemod --extensions jsx tsx
+  ```
+
+> [!WARNING]
+> Running the migration with all available extension options can possibly slow down the process. It is recommended to run the migration with only the necessary options.
````
I'm removing it, it's really slow in really big projects, shouldn't matter for most projects anyway

talkor replied:
It's required in the wizard, not in the command line. I'll remove the required. It already has these defaults

YossiSaadi replied:
ok got ya, cool

talkor commented on packages/codemod/README.md:
````diff
@@ -0,0 +1,70 @@
+  ```
+
+> [!WARNING]
+> Running the migration with all available extension options can possibly slow down the process. It is recommended to run the migration with only the necessary options.
````
I'm removing it, it's really slow in really big projects, shouldn't matter for most projects anyway

talkor commented on packages/codemod/README.md:
```diff
@@ -0,0 +1,70 @@
+
+- **Description**: Specifies the target directory where the transformations will be applied.
+- **Default**: Current working directory (`process.cwd()`)
+- **Required**: No (defaults to the current working directory if not provided)
```
Agree, removing

talkor commented on packages/core/src/__tests__/__snapshots__/exports.test.ts.snap:
```diff
@@ -46,8 +45,8 @@ exports[`exports should export all components 1`] = `
   "IconButton",
-  "InputField",
   "Label",
+  "LayerProvider",
```
Why not? We would be able to wrap other components like drawers and such, and get the same effect as in modals

YossiSaadi replied:
I need to review this again, so not sure I remember the purpose of it

talkor commented on packages/core/tsconfig.json:
```diff
@@ -3,7 +3,6 @@
   "compilerOptions": {
     "module": "ESNext",
     "declaration": true,
-    "declarationDir": "dist/types",
```
We used to export the types under the `monday-ui-react-core/types` entry, but now it's under the main entry

talkor commented on packages/core/src/storybook/stand-alone-documentaion/welcome/welcome.mdx:
```diff
@@ -40,3 +41,5 @@ Read our<Link href="https://github.com/mondaycom/vibe/blob/master/CONTRIBUTING.m
 
 <OtherContributorsList />
+
+<DocFooter feedbackFormLink="https://forms.monday.com/forms/213ebddcb0d423ae5b6178fb6e8f7b3d?r=use1" />
```
I removed it from all pages because it was a bit weird to have a feedback section in the migration guide and another this one as well. I guess I'll revert it for now and we'll think about it because I don't like it either

YossiSaadi replied:
agree

talkor commented on packages/core/.storybook/preview.tsx:
```diff
@@ -61,10 +61,7 @@ const preview: Preview = {
-            {children}
-            {<DocFooter feedbackFormLink="https://forms.monday.com/forms/213ebddcb0d423ae5b6178fb6e8f7b3d?r=use1" />}
-          </Unstyled>
+          <Unstyled>{children}</Unstyled>
```
Reverted

talkor commented on packages/icons/README.md:
I don't see it in the history, @rivka-ungar do you know maybe?

YossiSaadi replied:
I think I was mistaken, seem like there wasn't one in the PR
@rivka-ungar can you create a task to add a section about it?

YossiSaadi replied:
fixed by @rivka-ungar at #2581

talkor commented on packages/icons/scripts/generate-lazy-icons.js:
```diff
@@ -0,0 +1,41 @@
+const indexContent = iconFiles
+  .map(file => {
+    const fileNameWithoutExtension = path.basename(file, ".tsx");
+    return `export { default as Lazy${fileNameWithoutExtension}Icon } from './${fileNameWithoutExtension}';`;
```
@rivka-ungar?

YossiSaadi replied:
fixed by @rivka-ungar at #2581

talkor commented on packages/icons/src/svg/Redo.svg:
@rivka-ungar do you know maybe?

rivka-ungar replied:
It was accidentally deleted from styles and added as new instead of just moving with the rest of the icons. (Probably was added to vibe only after the other icons moved)

YossiSaadi commented:
just a short explanation on the latest commits here:

The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others

YossiSaadi commented:
> just a short explanation on the latest commits here:
> 
> The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others

Still fails, but a bit different issue.
For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.

YossiSaadi commented:
> > > just a short explanation on the latest commits here:
> > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > 
> > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> 
> Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI

Awesome you managed to get it to work!! 🥇

And I don't think we should remove them from the ignore, there's no need for them to be in the repo. 

We just need to make sure the build command cleans those folder, no?
Or I didn't understand correctly.

talkor commented:
> > > > just a short explanation on the latest commits here:
> > > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > > 
> > > 
> > > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> > 
> > 
> > Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> > 
> > 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> > 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI
 
> We just need to make sure the build command cleans those folder, no? Or I didn't understand correctly.

I don't think it's a good practice to clean a folder in the `src` folder on build 😕

YossiSaadi commented:
> > > > > just a short explanation on the latest commits here:
> > > > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > > > 
> > > > 
> > > > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> > > 
> > > 
> > > Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> > > 
> > > 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> > > 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI
>  
> > We just need to make sure the build command cleans those folder, no? Or I didn't understand correctly.
> 
> I don't think it's a good practice to clean a folder in the `src` folder on build 😕 
> 

It should be from the dist, why src?

talkor commented:
> > > > > > just a short explanation on the latest commits here:
> > > > > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > > > > 
> > > > > 
> > > > > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> > > > 
> > > > 
> > > > Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> > > > 
> > > > 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> > > > 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI
> > 
> > 
> > > We just need to make sure the build command cleans those folder, no? Or I didn't understand correctly.
> > 
> > 
> > I don't think it's a good practice to clean a folder in the `src` folder on build 😕
> 
> It should be from the dist, why src?

It's both in dist and src. The dist is not an issue, the folders in src are the troublemakers

rivka-ungar commented:
> > > > > just a short explanation on the latest commits here:
> > > > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > > > 
> > > > 
> > > > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> > > 
> > > 
> > > Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> > > 
> > > 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> > > 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI
> 
> > We just need to make sure the build command cleans those folder, no? Or I didn't understand correctly.
> 
> I don't think it's a good practice to clean a folder in the `src` folder on build 😕

They are folders that are created only on build, would it be bad to delete them right after the dist is created?

talkor commented:
> > > > > > just a short explanation on the latest commits here:
> > > > > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > > > > 
> > > > > 
> > > > > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> > > > 
> > > > 
> > > > Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> > > > 
> > > > 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> > > > 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI
> > 
> > 
> > > We just need to make sure the build command cleans those folder, no? Or I didn't understand correctly.
> > 
> > 
> > I don't think it's a good practice to clean a folder in the `src` folder on build 😕
> 
> They are folders that are created only on build, would it be bad to delete them right after the dist is created?

I think we should avoid having the build deleting folders from `src`, can we avoid creating them (and create only in the dist)?

rivka-ungar commented:
> > > > > > > just a short explanation on the latest commits here:
> > > > > > > The forced update of the package.json of codemods and icons without really releasing versions for the packages, alongside update to icons and snapshots from master caused the build to get stuck after its tests failed for Tipseen and others
> > > > > > 
> > > > > > 
> > > > > > Still fails, but a bit different issue. For some reason, only on CI, it gets the older version of the icons, and therefore the snapshot test fails, as it gets the old icon svg, instead of the new one. Very strange.
> > > > > 
> > > > > 
> > > > > Sorry @YossiSaadi I had to force push your recent commits, I found the issues:
> > > > > 
> > > > > 1. Non-fixated version `@vibe/icons` in `@vibe/core` - I changed the version to `1.0.0-rc.0` and removed the `^` as it shouldn't be present in the monorepo's packages.
> > > > > 2. Local issue with `@vibe/icon` - the `src/react` and `src/lazy` folders are ignored by git and are generated on build, and when working locally and switching branches (and running build but not necessarily), these folders are not updated, and so the monorepo takes these folders into account instead of the actual icons on that branch. This may causes some unwanted changes in some snapshots. We need to find a way to resolve it (I'll all for not gitignoring these folders as the overhead of dealing with them otherwise can be frustrating) - @rivka-ungar FYI
> > > 
> > > 
> > > > We just need to make sure the build command cleans those folder, no? Or I didn't understand correctly.
> > > 
> > > 
> > > I don't think it's a good practice to clean a folder in the `src` folder on build 😕
> > 
> > 
> > They are folders that are created only on build, would it be bad to delete them right after the dist is created?
> 
> I think we should avoid having the build deleting folders from `src`, can we avoid creating them (and create only in the dist)?

I remember that there was some issue when we tried that, I will look into it again
